third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -2739,6 +2740,116 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         }
     }
 
+    fn translate_fields_exp(
```

Comment:
This is adapted from `translate_fields`, but it additionally returns the local bindings introduced in the transformation.

Reply:
I have not fully reviewed this carefully, but on a first glance, this seems like a lot of repeated code from `translate_fields`. Please try refactoring (for example, can the same function be used to do both, using a boolen input to do uncommon stuff).

Reply:
I'm not quite sure why the same as translate_fields? Its actually calling it.